json/src/value.rs

```diff
@@ -327,6 +395,86 @@ impl Value {
     }
 }
 
+/// A type that can be used to index into a `serde_json::Value`. See the `get`
+/// and `get_mut` methods of `Value`.
+pub trait Index: private::Sealed {
```

Comment:
I got the technique from [here](https://www.reddit.com/r/rust/comments/25gvwl/prevent_another_crate_from_implementing_a_trait/) and the name from [here](https://internals.rust-lang.org/t/pre-rfc-sealed-traits/3108).